elastic/apm-server
# Fix agent_config metrics indexing

## Motivation/summary

Send `agent_config` metrics to the final batch processor chain, skipping authorization and pre-processing. Add a system test to verify the metrics are indexed. Also fixed a minor bug where we were reporting an `agent_config` metric with `labels.etag: "-"` for non-matching agent config queries.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~ only broken on main
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Configure agent config in Kibana
2. Set up an agent that would use that agent config
3. Wait 1-2 minutes
4. Refresh Kibana, check that agent config is marked as applied

## Related issues

Closes https://github.com/elastic/apm-server/issues/9048
